src/index.js

```diff
@@ -1,104 +1,105 @@
-  constructor(props) {
-    super(props);
-    this.state = {
-      imageLoaded: false,
-      isValidSrc: !!props.src,
-    };
-    this.showImage = (loadEvent = null) => {
-      // https://reactjs.org/docs/legacy-event-pooling.html
-      loadEvent !== null && loadEvent.persist();
+const RenderSmoothImage = ({
+  src,
+  alt,
+  objectFit,
+  imageProps,
+  wrapperProps,
+  onLoad,
+  onError
+}) => {
+  const [isValidSrc, setIsValidSrc] = useState(!!src);
+  const [imageLoaded, setImageLoaded] = useState(false);
+  const imageRef = useRef();
 
-      this.setState(
-        { imageLoaded: true },
-        () =>
-          loadEvent !== null &&
-          typeof props.onLoad === 'function' &&
-          props.onLoad(loadEvent)
-      );
-    };
-    this.handleError = (errorEvent = null) => {
-      // https://reactjs.org/docs/legacy-event-pooling.html
-      errorEvent !== null && errorEvent.persist();
-
-      this.setState(
-        { isValidSrc: false },
-        () =>
-          errorEvent !== null &&
-          typeof props.onError === 'function' &&
-          props.onError(errorEvent)
-      );
-    };
-    this.imageRef = React.createRef();
-  }
-
-  componentDidMount() {
-    if (this.state.isValidSrc) {
-      new Image().src = this.props.src;
+  useEffect(() => {
+    if (isValidSrc) {
+      new Image().src = src;
     }
     // Image tag is not rendered for invalid src - Hence the check for ref's presence.
-    if (!!this.imageRef.current && this.imageRef.current.complete) {
-      this.showImage();
+    if (!!imageRef.current?.complete) {
+      showImage();
     }
-  }
+  }, []);
+
+  const showImage = useCallback(
+    (loadEvent = null) => {
+      setImageLoaded(true);
+      if (loadEvent !== null && typeof onLoad === "function") onLoad(loadEvent);
+    },
+    [setImageLoaded]
+  );
 
-  render() {
-    const { imageLoaded, isValidSrc } = this.state;
-    const { src, alt, objectFit, imageProps, wrapperProps } = this.props;
+  const handleError = useCallback(
+    (errorEvent = null) => {
+      setIsValidSrc(false);
+      if (errorEvent !== null && typeof onError === "function")
+        onError(errorEvent);
+    },
+    [setIsValidSrc]
+  );
 
-    return (
-      <div className="smooth-image-wrapper" {...wrapperProps}>
-        {isValidSrc ? (
-          <img
-            ref={this.imageRef}
-            className={`smooth-image img-${imageLoaded ? 'visible' : 'hidden'}`}
-            style={{ objectFit }}
-            src={src}
-            alt={alt}
-            onLoad={this.showImage}
-            onError={this.handleError}
-            {...imageProps}
-          />
-        ) : (
-          <div className="smooth-no-image">{alt}</div>
-        )}
-        {isValidSrc && !imageLoaded && (
-          <div className="smooth-preloader">
-            <span className="loader" />
-          </div>
-        )}
-      </div>
-    );
-  }
-}
+  let imageClassnames = `smooth-image img-${
```

Comment:
This can be a `const`. I suppose the value is not changed anywhere.

Reply:
Resolved in https://github.com/KRRISH96/render-smooth-image-react/pull/10/commits/bfbcd7777d2b9560a2ef216666fa537d2aeedd9b